docs/search/idastar.md

```diff
@@ -30,56 +30,56 @@ Begin
 end;
 ```
 
-### 优点
+### Advantage
 
-1. 空间开销小，每个深度下实际上是一个深度优先搜索，不过深度有限制，而 DFS 的空间消耗小是众所周知的；
-2. 利于深度剪枝。
+1. The space overhead is small, and deep down is actually a depth-first search with limited depth. Also, it is well known that DFS consumes smaller space.
+2. Beneficial for deep pruning.
 
-### 缺点
+### Disadvantage
 
-重复搜索：回溯过程中每次 depth 变大都要再次从头搜索。
+Repetitive search: In the process of backtracking, each time the depth increases, the search must be done again from the beginning.
 
-> 其实，前一次搜索跟后一次相差是微不足道的。
+> In fact, the difference between the previous search and the next one is negligible.
 
-## 例题
+## Sample problem
 
-??? note "埃及分数"
-     **题目描述** 
+??? note "Egypt fraction"
+     **Problem description**
 
-    在古埃及，人们使用单位分数的和（即 $\frac{1}{a}$ ， $a$ 是自然数）表示一切有理数。例如， $\frac{2}{3}=\frac{1}{2}+\frac{1}{6}$ ，但不允许 $\frac{2}{3}=\frac{1}{3}+\frac{1}{3}$ ，因为在加数中不允许有相同的。
+    In ancient Egypt, people use the sum of unit fractions (that is, $\frac{1}{a}$ , $a$ is a natural number) to represent all rational numbers. For example, $\frac{2}{3}=\frac{1}{2}+\frac{1}{6}$ but $\frac{2}{3}=\frac{1}{ 3}+\frac{1}{3}$ is not allowed because same number cannot be added.
 
-    对于一个分数 $\frac{a}{b}$ ，表示方法有很多种，其中加数少的比加数多的好，如果加数个数相同，则最小的分数越大越好。例如， $\frac{19}{45}=\frac{1}{5}+\frac{1}{6}+\frac{1}{18}$ 是最优方案。
+    For a fraction $\frac{a}{b}$ , there are many ways to represent it. Among them, the one with less addends is better than the one with more addends. If the number of additions is the same, the larger the minimum fraction, the better. For example, $\frac{19}{45}=\frac{1}{5}+\frac{1}{6}+\frac{1}{18}$ is the optimal solution.
```

Comment:
Among them, the one with less operands is preferred, and when the number of operands equals, the larger ..., the better.